### PR TITLE
Packages: update versioning and release process

### PR DESCRIPTION
**Pull request by dprokop**
@ryantxu @torkelo

This PR introduces updated Readmes that summarise guidelines about packages versioning and releases. Also, ALPHA notices are added to every package's Readme. 

---

I have been thinking recently about the approach to versioning packages. This seems to be slightly tricky, especially that we have some experience now with loads of versions pushed to npm.

Here are my thoughts:
1. Packages' version should ALWAYS reflect current Grafana version
2. We should stop using `alpha` prerelease id in every version of packages. **I think having an an explicite warnings about packages being in ALPHA state until further notice is sufficient**. Having alpha in every version on npm is making managing versioning more complicated than it should be and it doesn't bring us any value and it's a pain to say which version of package corresponds to which version of Grafana.
3. We should be publishing packages under 3 different dist-tags on npm:
   - `latest` - version that is released with latest stable Grafana, i.e. 6.2.5
   - `next` - version that is released with current Grafana prerelease, i.e. 6.3.0.beta.1
   - `canary` - version that is released directly from master on every merge, i.e. 6.4.0-pre-bbc3e10b. My main motivation for `canary` is that master isn't actually the next version. It's a *possible* next version, depending on the commits that's gonna be cherry picked to the release branch. Also, what is very important, `lerna.json` version on master should always be the same as in package.json, meaning it's the next major/minor i.e. `6.4.0-pre`. The `pre` doesn't seem like semantic versioning, but as long as we are consistent, this should not be an issue. Grafana doesn't strictly follow semver anyways.

___

**So, some action points if this is going to be merged and followed:**
- [x] Update `scripts/circle-release-next-packages.sh` to publish to `canary` dist-tag instead of `next`:
   - Either introduce `packages:publishCanary` script
   - or use `yarn packages:publish --dist-tag canary --yes` in the script
- [x] Sync with https://github.com/grafana/grafana/pull/18173 (update versions to `6.4.0-pre`)
- [x] Next Grafana release will include packages release as well. Depending whether its gonna be beta.2 or stable, we should publish to npm with version corresponding to Grafana version. This is introduced in eb32fe569a15ec3b9416cd30f823148a38e70d60
- [x] Grafana Release Guide should be updated with new additional step that would describe the process or redirect to [this guide](https://github.com/grafana/grafana/blob/bbc3e10b2f3b3d3570c84b4e6a6db89319882365/packages/README.md#manual-release). I suggest that you @ryantxu assist anyone who's gonna do the release
- [ ] Since there are a lot of versions published to npm already I thought it would be wise to cleanup the history there a little bit. But, if we skip adding alpha to every version, and follow Grafana's versioning, I think we should be fine with those early versions being published on npm. We can ask them to remove some of them later on.
- [ ] ~We should consider publishing from master (canary release according to ^^) to different registry than npm and maybe treat npm as a source of packages released with Grafana. This is described in https://github.com/grafana/grafana/issues/18038#issuecomment-512797352 . Thanks to that we could avoid having a lot of noise in our npm history.~ We don't have access to GH package registry yet.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
Looks good!"

**Review by ryantxu — APPROVED** [review]
Looks good -- I'll wait for you to get back to pull the trigger :)

File: packages/README.md [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
+> Make sure you are logged in to npm in your terminal and that you are a part of Grafana org on npm
+
+1. Run `yarn packages:prepare` script from root directory. This will perform tests on the packages and prompt for version of the packages. The version should be the same as the one being released.
+  - Make sure you use semver convention. So, *place a dot between prerelease id and prelease number*!!! i.e. 6.3.0-alpha.1
+  - Make sure you confirm the version bump when prompted!
+2. Commit changes (lerna.json & package.json files) - *"Packages version update: \<VERSION\>"*
+3. Run `yarn packages:build` script that will prepare distribution packages in `packages/grafana-*/dist`. These directories are going to be published to npm
+4. Depending whether  or not it's a prerelease:
+  - When releasing a prelease run `packages:publishNext` to publish new versions.
+  - When releasing a stable version run `packages:publish` to publish new versions.
```
Comment (dprokop) [inline packages/README.md]:
```suggestion
  - When releasing a stable version run `packages:publishLatest` to publish new versions.
```

**Comment by dprokop** [comment]
@ryantxu @torkelo can u review one more time, pushed some commits there with script to release next/latest versions when releasing Grafana.

File: scripts/build/release-packages.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (xlson) [inline scripts/build/release-packages.sh]:
```suggestion
#!/usr/bin/env bash
```

File: scripts/build/release-packages.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+if grep -q "beta" <<< "$GRAFANA_TAG"; then
+  RELEASE_CHANNEL="next"
+fi
+
+echo "$_grafana_version"
+
+# Get current version from lerna.json
+# Since this happens on tagged branch, the lerna.json version and package.json file SHOULD be updated already
+# as specified in release guideline
+PACKAGE_VERSION=$(grep '"version"' lerna.json | cut -d '"' -f 4)
```
Comment (xlson) [inline scripts/build/release-packages.sh]: This expects `lerna.json` to be in your work dir, but as you `cd` into `scripts/build` before running this script that won't be the case. You could take in the path to lerna.json as an arg or figure out the correct relative path or just run the script from the root of the project.

File: scripts/build/release-packages.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  SCRIPT="publishLatest"
+elif [ $RELEASE_CHANNEL == "next" ]; then
+  SCRIPT="publishNext"
+else
+  echo "Unknown channel, skipping packages release"
+  exit
+fi
+
+echo $'\nBuilding packages'
+yarn packages:build
```
Comment (xlson) [inline scripts/build/release-packages.sh]: I presume the yarn commands will fail for the same reason as lerna.json. This expects the workdir to be project root, right?

File: scripts/build/release-packages.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+echo $'\nPublishing packages'
+yarn packages:${SCRIPT}
```
Comment (xlson) [inline scripts/build/release-packages.sh]: same

File: scripts/build/release-packages.sh (hunk 3 above)
Comment (dprokop) [reply scripts/build/release-packages.sh]: Yeah, you are right!

File: .circleci/config.yml [hunk 6]
```diff
@@ -675,6 +675,21 @@ jobs:
...
+      - checkout
+      - run:
+          name: Boostrap lerna
+          command: 'npx lerna bootstrap'
+      - run:
+          name: npm - Prepare auth token
+          command: 'echo //registry.npmjs.org/:_authToken=$NPM_TOKEN >> ~/.npmrc'
+      - run:
+          name: Release packages
+          command: cd scripts/build && ./release-packages.sh "${CIRCLE_TAG}"
```
Comment (dprokop) [inline .circleci/config.yml]:
```suggestion
          command: ./scripts/build/release-packages.sh "${CIRCLE_TAG}"
```

File: scripts/build/release-packages.sh (hunk 4 above)
Comment (dprokop) [reply scripts/build/release-packages.sh]: Fixed in 99a8244514af2d7710b38adbceae95a31babe460

File: scripts/build/release-packages.sh (hunk 3 above)
Comment (dprokop) [reply scripts/build/release-packages.sh]: Fixed in 99a8244514af2d7710b38adbceae95a31babe460

File: scripts/build/release-packages.sh (hunk 5 above)
Comment (dprokop) [reply scripts/build/release-packages.sh]: Fixed in 99a8244514af2d7710b38adbceae95a31babe460

**Review by xlson — APPROVED** [review]
LGTM!

**Review by ryantxu — APPROVED** [review]
LGTM too!

**Pull request merged**